test/Reflection/ParameterReflectionTest.php

```diff
@@ -122,7 +124,12 @@ public function testGetType(
 
         $type = $reflection->getType();
 
+        if (PHP_VERSION_ID >= 80500 && $expectedType == 'self') {
+            $expectedType = $className;
+        }
```

Comment:
Not only are tests not running on 8.5 yet, but I think the version comparison is unnecessary and can be changed to:

```php
if ($expectedType === 'self') {
   $expectedType = $className;
}
```